internal/testrunner/runners/system/runner.go

```diff
@@ -379,6 +375,11 @@ func (r *runner) runTest(config *testConfig, ctxt servicedeployer.ServiceContext
 		return nil
 	}
 
+	logger.Debug("assigning package data stream to agent...")
+	if err := kib.AssignPolicyToAgent(agent, *policy); err != nil {
+		return result.WithError(errors.Wrap(err, "could not assign policy to agent"))
+	}
+
```

Comment:
Just curious, why did this block need to be moved?

Reply:
Assigning a policy is a long operation (due to condition "wait for the policy to get assigned"). If you stop here, but in the mean time an agent gets assigned, you won't be able to delete policy. Just in case it's safer to reassign the original policy.